store/tikv/coprocessor.go

```diff
@@ -1189,3 +1206,144 @@ func (it copErrorResponse) Next(ctx context.Context) (kv.ResultSubset, error) {
 func (it copErrorResponse) Close() error {
 	return nil
 }
+
+type rateLimitAction struct {
+	enabled        bool
+	fallbackAction memory.ActionOnExceed
+	once           sync.Once
+	// totalTokenNum indicates the total token at initial
+	totalTokenNum uint
+	cond          struct {
+		*sync.Cond
+		// exceed indicates whether have encountered OOM situation.
+		exceed bool
+		// existedTokenNum indicates the count of tokens which still exists
+		existedTokenNum uint
+		// isTokenDestroyed indicates whether there is one token has been isTokenDestroyed after Action
+		isTokenDestroyed bool
+		// maxID indicates the max id of the running copTask
+		maxID int
+	}
+}
+
+func newrateLimitAction(sendRate *rateLimit, cond *sync.Cond) *rateLimitAction {
```

Comment:
1. newRateLimitAction
2. We'd better pass `uint(cap(sendRate.token))` instead of `sendRate`